### ISSUE TITLE
[clsim] AutoSetGeant4Environment.py assumes that I3_PORTS is set (Trac #2081)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2073]
This script assumes that `$I3_PORTS` is a meaningful path, which is not true in the py2-v3 environment. Mostly it seems to need to find `geant4.sh`, but it looks as though this should be trivial since this should always be in `$PATH`. I am less sure about how the entries in `hardCodedForGeant4_9_3and4` should be handled, but it looks as though these should be found relative to the location of `geant4.sh`. 

Migrated from https://code.icecube.wisc.edu/ticket/2081
```json
{
    "status": "closed", 
    "changetime": "2019-03-22T12:58:54", 
    "description": "This script assumes that `$I3_PORTS` is a meaningful path, which is not true in the py2-v3 environment. Mostly it seems to need to find `geant4.sh`, but it looks as though this should be trivial since this should always be in `$PATH`. I am less sure about how the entries in `hardCodedForGeant4_9_3and4` should be handled, but it looks as though these should be found relative to the location of `geant4.sh`. ", 
    "reporter": "cweaver", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1553259534799042", 
    "component": "combo simulation", 
    "summary": "[clsim] AutoSetGeant4Environment.py assumes that I3_PORTS is set", 
    "priority": "normal", 
    "keywords": "CLSim", 
    "time": "2017-08-17T23:31:13", 
    "milestone": "Vernal Equinox 2019", 
    "owner": "claudio.kopper", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170818T16:02:39: david.schultz changed milestone from "" to "December 2017"

**Comment by icecube-trac** [comment]
Trac update at 20180607T16:32:25: olivas changed milestone from "December 2017" to "Autumnal Equinox 2018"

**Comment by icecube-trac** [comment]
Trac update at 20181001T20:49:48: olivas changed summary from "AutoSetGeant4Environment.py assumes that I3_PORTS is set" to "[clsim] AutoSetGeant4Environment.py assumes that I3_PORTS is set"

**Comment by icecube-trac** [comment]
Trac update at 20181002T15:41:48: olivas commented:

> The first couple lines of that file:
> 
> # FIXME: The functionality of this script is now integrated into env-shell.sh,
> #        therefore this script should be removed at some point to avoid confusion.
> 
> I3CLSimMakePhotons.py and tabulator.py still use it.
> 
> In I3CLSimMakePhotons.py:
>     # at the moment the Geant4 paths need to be set, even if it isn't used
>     # TODO: fix this
>     if clsim.I3CLSimLightSourceToStepConverterGeant4.can_use_geant4:
>         AutoSetGeant4Environment()
> 
> ...and in tabulator.py:
>     # at the moment the Geant4 paths need to be set, even if it isn't used
>     # TODO: fix this
>     if clsim.I3CLSimLightSourceToStepConverterGeant4.can_use_geant4:
>         AutoSetGeant4Environment()
> 
> Going to kick this can down the road and continue with triage.

**Comment by icecube-trac** [comment]
Trac update at 20181008T22:37:05: olivas changed milestone from "Autumnal Equinox 2018" to "Winter Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20190322T12:58:54: 
* olivas changed status from "new" to "closed"
* olivas commented:

> In changeset 172252:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="172252"
> uh...we fixed the glitch.  we find it's best to do these things on a friday. closes #2073
> ```


* olivas changed resolution from "" to "fixed"
